dor/providers/models.py

```diff
@@ -0,0 +1,71 @@
+from dataclasses import dataclass, field
+import uuid
+from datetime import datetime
+from enum import Enum
+
+@dataclass
+class AlternateIdentifier:
+    type: str
+    id: str
+
+
+@dataclass
+class Agent:
+    address: str
+    role: str
+
+
+@dataclass
+class PreservationEvent:
+    identifier: str
+    type: str
+    datetime: datetime
+    detail: str
+    agent: Agent
+
+
+@dataclass
+class FileReference:
+    locref: str
+    mdtype: str = None
+    mimetype: str = None
+
+
+@dataclass
+class FileMetadata:
+    id: str
+    use: str
+    mdid: str = None
+    groupid: str = None
+    ref: FileReference = None
```

Comment:
Suggestion: I'd evaluate whether some of these variables could be made a bit more descriptive, specifically or especially `locref`, `mdtype`, and `mdid`. `locref` seems to be location or URI, but I don't really know what the others are.

Reply:
I will insist we don't have enough of the full DOR picture to bother with finalizing names. We _could_ end up with a structure like 

```
"file_identifiers": [
        {
          "id": "disk:///Users/roger/Projects/objectclass/figgy/tmp/more_files/54/96/b5/5496b5885f8140c8a386a9171646a303/example.tif"
        }
      ],
```

where we "resolve" IDs (no schema == internal resource). We just don't know yet.